test/local.jl

```diff
@@ -34,6 +35,26 @@ using TimeZones: TimeZone, localzone, parse_tz_format
 
 
 @testset "localzone" begin
+    @testset "_path_tz_name" begin
+        @test _path_tz_name("") === nothing
+        @test _path_tz_name("/tmp/UTC/file") === nothing
+
+        for name in ("UTC", "Europe/Warsaw", "America/Indiana/Indianapolis")
+            # Use eval to improve readability of tests when failures occur
+            @eval begin
+                @test _path_tz_name($name) == $name
+                @test _path_tz_name("/usr/share/zoneinfo/$($name)") == $name
+                @test _path_tz_name("/var/db/timezone/zoneinfo/$($name)") == $name
+            end
+        end
+
+        # The algorithm that extracts the tz name could accidentally return the valid
```

Comment:
Is this expected behaviour or are we testing that this doesn't happen anymore? Comment could use clarification either way

Reply:
I'll expand on this comment

Reply:
Let me know if the new comment is clearer